### PR TITLE
OpenCV: New versions, adjust protobuf dependency

### PR DESCRIPTION
**Pull request by JSquar**
* New versions of OpenCV are available
* Removed protobuf dependency since opencv builds its own. Having a dependency on protobuf leads to header mismatch if spack's protobuf is included but opencv has an older version of protobuf
* Cuda conflicts gcc > 7 (`/home/squar/Git/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-8.3.0/cuda-10.0.130-4jz6namrc73qrjfik5qevef5jiih73xn/include/crt/host_config.h:129:2: error: #error -- unsupported GNU version! gcc versions later than 7 are not supported!`). By the way, I'm suprised there is nothing inside the cuda package even though cuda is quite picky with compatible compiler versions in general
* Because of other issues, which have nothing to do with openCV itself, some variants/dependencies have to be set. I successfully built openCV with `spack install opencv~cuda^vtk+osmesa@8.0.1^hdf5+hl^python@3:` with gcc@8.3.0 on Ubuntu@18.04

### CODE REVIEW

File: var/spack/repos/builtin/packages/opencv/package.py [hunk 1]
```diff
@@ -107,7 +118,6 @@ class Opencv(CMakePackage):
     depends_on('qt', when='+qt')
     depends_on('java', when='+java')
     depends_on('py-numpy', when='+python', type=('build', 'run'))
-    depends_on('protobuf@3.1.0', when='@3.3.0: +dnn')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/opencv/package.py]: Is there a way to tell opencv not to build its own protobuf? Allowing packages to build their dependencies is entirely against the Spack model and often causes dependency incompatibility problems when we compose dependencies into larger packages.

Comment (JSquar) [reply var/spack/repos/builtin/packages/opencv/package.py]: You're right. I found a corresponding flag and added it to the package. Therefore, I reinserted the dependency again (considering that the current opencv version requires another protobuf version)

**Comment by JSquar** [comment]
Just an annotation: I could test my changes only on opencv@4.1.0, because building older versions (like `spack install opencv@3.4.0~cuda^vtk+osmesa@8.0.1^hdf5+hl^python@3:`) fails on my system:
```
1 error found in build log:
     59    -- Detecting C compile features
     60    -- Detecting C compile features - done
     61    -- Detected version of GNU GCC: 83 (803)
     62    -- Performing Test HAVE_CXX11 (check file: cmake/checks/cxx11.cpp)
     63    -- Performing Test HAVE_CXX11 - Success
     64    -- Found PythonInterp: /usr/bin/python2.7 (Required is at least version "2.7")
  >> 65    CMake Error at cmake/OpenCVDetectPython.cmake:59 (if):
     66      if given arguments:
     67    
     68        "2" "EQUAL"
     69    
     70      Unknown arguments specified
     71    Call Stack (most recent call first):
```

But this does not correspond to my changes, it also happens with an unmodified package. I looked into the source of opencv to retrieve the version range of protobuf, so the dependency range should (famous last words) be right.

**Comment by scheibelp** [comment]
> But this does not correspond to my changes

Agreed, that can be addressed later.

**Pull request merged**